File: Vibe.Domain/SupportRequests/SupportRequestDetail.cs
using Vibe.Domain.Clients;
using Vibe.Domain.Employees;
using Vibe.Domain.SupportRequests.SupportMessages;

namespace Vibe.Domain.SupportRequests
{
    public class SupportRequestDetail
    {
        public Guid Id { get; }
        public String Title { get; }
        public String Description { get; }
        public Client Client { get; }
        public Employee? Employee { get; }
        public DateTime OpenedAt { get; set; }
        public Boolean IsClosed { get; }
        public SupportMessage[] Messages { get; }

        public SupportRequestDetail(Guid id, String title, String description, Client client, Employee? employee,
            DateTime openedAt, Boolean isClosed, SupportMessage[] messages)
        {
            Id = id;
            Title = title;
            Description = description;
            Client = client;
            Employee = employee;
            OpenedAt = openedAt;
            IsClosed = isClosed;
            Messages = messages;
        }
    }
}

File: Vibe.BackOffice/vibe.backoffice.client/src/domain/techSupports/supportRequestDetail.ts
import { Client, mapToClient } from "../clients/client";
import { SupportMessage, mapToSupportMessages } from "./supportMessage";

export class SupportRequestDetail {
    constructor(
        public id: string,
        public title: string,
        public description: string,
        public client: Client,
        public employeeId: string | null,
        public openedAt: Date,
        public isClosed: boolean,
        public messages: SupportMessage[] = []
    ) { }
}

export function mapToSupportRequestDetail(data: any): SupportRequestDetail {
    const messages: SupportMessage[] = mapToSupportMessages(data.messages)

    const client = mapToClient(data.client)
    
    return new SupportRequestDetail(data.id, data.title, data.description, client, data.employeeId,
        new Date(data.openedAt), data.isClosed, messages)
}